File: packages/java/parser-jvm-plugin-nonnull/src/test/java/com/vaadin/fusion/parser/plugins/nonnull/extended/ExtendedEndpoint.java
package com.vaadin.fusion.parser.plugins.nonnull.extended;

import java.util.List;
import java.util.Map;

import com.vaadin.fusion.parser.plugins.nonnull.basic.BasicEndpoint;

@Endpoint
public class ExtendedEndpoint extends BasicEndpoint {
    public List<Map<String, List<Map<String, @Nonnull String>>>> superComplexType(
            List<Map<String, List<Map<String, @Nonnull String>>>> list) {
        return list;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/extended/snapshots/ExtendedEndpoint.ts
import type { EndpointRequestInit } from '@vaadin/hilla-frontend';
import type Entity from './com/vaadin/hilla/parser/plugins/nonnull/extended/ExtendedEndpoint/Entity.js';
import client from './connect-client.default.js';

export async function getNonnullListOfNullableElements(init?: EndpointRequestInit): Promise<Array<Entity | undefined>> {
  return client.call('ExtendedEndpoint', 'getNonnullListOfNullableElements', {}, init);
}

export async function superComplexType(
  list: Array<Record<string, Array<Record<string, string> | undefined> | undefined> | undefined> | undefined,
  init?: EndpointRequestInit,
): Promise<Array<Record<string, Array<Record<string, string> | undefined> | undefined> | undefined> | undefined> {
  return client.call('ExtendedEndpoint', 'superComplexType', { list }, init);
}
